esp32_s3_lcd_ev_board/include/bsp/esp32_s3_lcd_ev_board.h

```diff
@@ -0,0 +1,369 @@
+#define BSP_LCD_DATA3           (GPIO_NUM_13)   // B6
+#define BSP_LCD_DATA4           (GPIO_NUM_14)   // B7
+#define BSP_LCD_DATA5           (GPIO_NUM_21)   // G2
+#define BSP_LCD_DATA6           (GPIO_NUM_47)   // G3
+#define BSP_LCD_DATA7           (GPIO_NUM_48)   // G4
+#define BSP_LCD_DATA8           (GPIO_NUM_45)   // G5
+#define BSP_LCD_DATA9           (GPIO_NUM_38)   // G6
+#define BSP_LCD_DATA10          (GPIO_NUM_39)   // G7
+#define BSP_LCD_DATA11          (GPIO_NUM_40)   // R3
+#define BSP_LCD_DATA12          (GPIO_NUM_41)   // R4
+#define BSP_LCD_DATA13          (GPIO_NUM_42)   // R5
+#define BSP_LCD_DATA14          (GPIO_NUM_2)    // R6
+#define BSP_LCD_DATA15          (GPIO_NUM_1)    // R7
+#define BSP_LCD_SPI_CS          (IO_EXPANDER_PIN_NUM_1)
+#define BSP_LCD_SPI_SCK         (IO_EXPANDER_PIN_NUM_2)
+#define BSP_LCD_SPI_SDO         (IO_EXPANDER_PIN_NUM_3)
+#endif
+
+/* USB */
+#define BSP_USB_POS             (USBPHY_DP_NUM)
+#define BSP_USB_NEG             (USBPHY_DM_NUM)
+
+/* Button */
+typedef enum {
+    BSP_BUTTON_BOOT = GPIO_NUM_0,
+} bsp_button_t;
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**************************************************************************************************
+ *
+ * I2C Interface
+ *
+ * There are multiple devices connected to I2C peripheral:
+ *  - Codec ES8311 (configuration only)
+ *  - ADC ES7210 (configuration only)
+ *  - LCD Touch controller
+ *  - IO expander chip TCA9554
+ *
+ * After initialization of I2C, use `BSP_I2C_NUM` macro when creating I2C devices drivers ie.:
+ * \code{.c}
+ * es8311_handle_t es8311_dev = es8311_create(BSP_I2C_NUM, ES8311_ADDRRES_0);
+ * \endcode
+ *
+ **************************************************************************************************/
+#define BSP_I2C_NUM             (CONFIG_BSP_I2C_NUM)
+
+/**
+ * @brief Init I2C driver
+ *
+ * @return
+ *      - ESP_OK:               On success
+ *      - ESP_ERR_INVALID_ARG:  I2C parameter error
+ *      - ESP_FAIL:             I2C driver installation error
+ *
+ */
+esp_err_t bsp_i2c_init(void);
+
+/**
+ * @brief Deinit I2C driver and free its resources
+ *
+ * @return
+ *      - ESP_OK:               On success
+ *      - ESP_ERR_INVALID_ARG:  I2C parameter error
+ *
+ */
+esp_err_t bsp_i2c_deinit(void);
+
+/**************************************************************************************************
+ *
+ * IO Expander Interface
+ *
+ **************************************************************************************************/
+#define BSP_IO_EXPANDER_I2C_ADDRESS     (ESP_IO_EXPANDER_I2C_TCA9554_ADDRESS_000)
+
+/**
+ * @brief Init IO expander chip TCA9554
+ *
+ * @note If the device was already initialized, users can also call it to get handle
+ * @note This function will be called in `bsp_display_start()` when using LCD sub-board 2 with the resolution of 480x480
+ *
+ * @return Pointer to device handle or NULL when error occured
+ */
+esp_io_expander_handle_t bsp_io_expander_init(void);
```

Comment:
Hm, usually we don't provide wrapper functions around dependencies of BSP. The users can init the expander them selves by calling `esp_io_expander_new_i2c_tca9554`.

The idea is that if we provide only the `init` wrapper, the API look incomplete (the rest of the API is defined in tca9554 component).

What do you think?

_EDIT:_ Now I see that you need to use the expander to enable the audio amplifier... who should own the expander handle then?

Reply:
And if sub_board_480x480 is used, it's necessary to init tca9554, which you can see [here](https://github.com/espressif/esp-bsp/pull/89/commits/abc7099df435b938ce2cd02f1ecafc7ca546c79c#diff-521a0f3ed3e2e3565f9da746eb3de4eed006957460f7cd6077b1491d48d90f60R278-R280). So I left API `bsp_io_expander_init` for users to init tca9554 or just get the handle of it.

Reply:
OK, thank you